eppoclient/configurationstore.go

```diff
@@ -46,29 +43,18 @@ func newConfigurationStore(maxEntries int) *configurationStore {
 }
 
 func (cs *configurationStore) GetConfiguration(key string) (expConfig experimentConfiguration, err error) {
-	value, _ := cs.cache.Get(key)
-
-	if value == nil {
-		err = errors.New("not found")
-		return
-	}
-
-	jsonString, err := json.Marshal(value)
-
-	if err != nil {
-		log.Fatalln("Incorrect json")
-	}
-	ec := experimentConfiguration{}
-	err = json.Unmarshal(jsonString, &ec)
-	if err != nil {
-		log.Fatalln("failure to unmarshal json into experiment configuration")
+	// Attempt to get the value from the cache
+	expConfig, ok := cs.cache.Get(key)
```

Comment:
`expConfig` is already an instantiate type, no need to deal with intermediate strings or json. There is a bit of golang short-hand here, the output variable is declared as `expConfig experimentConfiguration` so it's already type checked against that.